doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,345 @@
+    the product
+  
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. This requires the board
+to:
+
+  - Perform its promised task reliably
+  - Integrate and communicate reliably with the main processor
+  - Include and support any relevant hardware
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+This requires:
+
+  - The presence of didactic materials related to RIOT
+  - Clear and easy to find documentation
+  - A general availability of supported hardware and tools
+
+# 3. Design philosophies
+
+RIOT satisfies the requirements of all the use cases given above, and does so
+in its own unique way, a way that reflects the collective personality of the
+RIOT community. RIOT is the Friendly IoT Operating System: it is friendly to
+users, allowing them to easily build and deploy IoT solutions that are stable
+and trouble-free.
+
+Below are the design philosophies that are typically followed by developers to
+achieve this, including descriptions of tradeoffs between them and where the
+resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't collecting data.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler. They shouldn't demand that users do
+the same.  They should, however, allow users to manage power through different
+modes and functions.
+
+#### Small memory footprint
+
+Most of RIOT's targeted use cases are well addressed by devices in class 1 of
+the taxonomy presented in [1]. If small price differences are important or the
+energy budget is particularly tight, the available memory might be near the
+bottom of this class. Over-the-air updating currently reduces the available ROM
+by over half.
+
+RIOT should provide out-of-the-box support for devices with ~10 KiB of
+available RAM and ~100 KiB of ROM. It should be just as possible to address
+real use cases involving even smaller devices, starting at ~2 KiB of RAM and
+~10 KiB of ROM, with manual optimizations and specialised modules. Users should
+be given the choice of what they want to spend their memory on.
+
+#### Constrained networking
+
+RIOT should deliver best-in-class communication bandwidth and robustness.
+
+Network stacks should remain up-to-date as relevant standards emerge; they
+should be adequately extensible to support this. Users should be able to adapt
+them and choose between different stacks; they should be able to get the best
```

Comment:
I find this statement difficult: "Users should be able to adapt them and choose between different stacks;"
This is for two reasons: 
 - What does adapt mean? Do you mean "configure according to technological options"?
 - Choosing between different protocol implementations is fine, if they are available. However, I don't think it is an objective of RIOT to provide multiple implementations of all networking protocols. Rather, we should be happy to provide a large variety of networking options at a high level of quality.

Reply:
Point1: yes; point2: I agree, I have clarified this statement